addons/misra.py

```diff
@@ -3157,15 +3157,15 @@ def misra_16_3(self, rawTokens):
         STATE_OK = 2  # a case/default is allowed (we have seen 'break;'/'comment'/'{'/attribute)
         STATE_SWITCH = 3  # walking through switch statement scope
 
-        define = None
+        macro = None
         state = STATE_NONE
         end_switch_token = None  # end '}' for the switch scope
         for token in rawTokens:
-            if simpleMatch(token, '# define'):
-                define = token
-            if define:
-                if token.linenr != define.linenr:
-                    define = None
+            if simpleMatch(token, '# define') or simpleMatch(token, '# pragma'):
+                macro = token
```

Comment:
the code now says that pragma is a "macro" I feel that is unfortunate. Maybe "directive" would be better.